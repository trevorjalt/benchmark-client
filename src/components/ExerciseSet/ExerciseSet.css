/* .ExerciseSet__item {
    display: flex;
    justify-content: space-between;
} */

.SetItemUpdateForm {
    display: flex;
    justify-content: center;
}

input {
  border: 1px solid #ddd;
  /* display: block; */
  font-size: 1.1em;
  margin-bottom: 10px; 
  padding: 1vh 0 1vh 0;
  text-align: left;
  /* width: 95vw; */
  border-radius: 4px;
}

.Input::placeholder {
  color: #ccc;
}

