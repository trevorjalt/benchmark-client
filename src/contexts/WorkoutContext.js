import React, { Component } from 'react'

const WorkoutContext = React.createContext({
    workoutList: [],
    exerciseList: [],
    exerciseSetList: [],
    newWorkoutItem: {},
    error: null,
    isLoggedIn: null,
    clearError: () => {},
    setError: () => {},
    setIsLoggedIn: () => {},
    setWorkoutList: () => {},
    setExerciseList: () => {},
    onDeleteExercise: () => {},
    onDeleteExerciseSet: () => {},
    onDeleteWorkout: () => {},
    onUpdateExercise: () => {},
    onUpdateExerciseSet: () => {},
})
export default WorkoutContext

export class WorkoutProvider extends Component {
    state = {
        workoutList: [],
        exerciseList: [],
        exerciseSetList: [],
        newWorkoutItem: {},
        error: null,
        isLoggedIn: null,
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    setIsLoggedIn = (value) => {
        this.setState({ isLoggedIn: value })
    }

    setWorkoutList = workoutList => {
        this.setState({ workoutList})
    }

    setExerciseList = exerciseList => {
        this.setState({ exerciseList})
    }

    setExerciseSetList = exerciseSetList  => {
        this.setState({ exerciseSetList })
    }

    onDeleteExercise = (newList) => {
        this.setExerciseList(newList)
    }

    onDeleteExerciseSet = (newList) => {
        this.setExerciseSetList(newList)
    }

    onDeleteWorkout = (newList) => {
        this.setWorkoutList(newList)
    }

    onUpdateExercise = (newList) => {
        this.setExerciseList(newList)
    }

    onUpdateExerciseSet = (newList) => {
        this.setExerciseSetList(newList)
    }

    render() {
        const value = {
            workoutList: this.state.workoutList,
            exerciseList: this.state.exerciseList,
            exerciseSetList: this.state.exerciseSetList,
            newWorkoutItem: this.state.newWorkoutItem,
            error: this.state.error,
            isLoggedIn: this.state.isLoggedIn,
            touched: this.state.touched,
            clearError: this.clearError,
            setError: this.setError,
            setIsLoggedIn: this.setIsLoggedIn,
            setWorkoutList: this.setWorkoutList,
            setExerciseList: this.setExerciseList, 
            setExerciseSetList: this.setExerciseSetList,
            onDeleteExercise: this.onDeleteExercise,
            onDeleteExerciseSet: this.onDeleteExerciseSet,
            onDeleteWorkout: this.onDeleteWorkout,
            onUpdateExercise: this.onUpdateExercise,
            onUpdateExerciseSet: this.onUpdateExerciseSet,         
        }
        return (
            <WorkoutContext.Provider value={value}>
                {this.props.children}
            </WorkoutContext.Provider>
        )
    }
}