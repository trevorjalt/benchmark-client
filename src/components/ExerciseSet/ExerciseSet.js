import React, { Component } from 'react'
import WorkoutContext from '../../contexts/WorkoutContext'
import { Button } from '../Utils/Utils'
import './ExerciseSet.css'


export default class ExerciseSet extends Component {
    static contextType = WorkoutContext

    // state = {
    //     set_weight: 0,
    //     set_repetition: 0,
    // }

    onWeightInputChange = event => {
        console.log(this.props.exerciseSet.id)
        this.props.onWeightChange(this.props.exerciseSet.id, Number(event.target.value), this.props.exerciseSet.exercise_id)
        // this.setState({ set_repetition: event.target.value})
    }
    
    onRepetitionInputChange = event => {
        this.props.onRepetitionChange(this.props.exerciseSet.id, Number(event.target.value), this.props.exerciseSet.exercise_id)
        // this.props.handleUpdateSet(this.state.set_repetition)
    }

    // handleFormChange(event) {
    //     const { exerciseSet } = this.props
    //     const updateSet = {
    //         id: exerciseSet.id,
    //         set_weight: this.state.set_weight,
    //         set_repetition: this.state.set_repetition,
    //     }
    //     this.props.handleUpdateSet(updateSet)
    //     // this.forceUpdate()
    // }

    // handleUpdateChange = () => {
    //     const updateSet = { set_weight: this.state.set_weight, set_repetition: this.state.set_repetition }
    //     this.props.handleUpdateSet(updateSet)
    //     // const updateSet =  {this.handleWeightChange
    // }

    // handleClickUpdate = event => {
    //     event.preventDefault()
    //     const { exerciseSet, edit } = this.props
    //     const updatedExerciseSet = {
    //         id: exerciseSet.id,
    //         // set_weight: Number(event.target.SetItem__weight.value),
    //         // set_repetition: Number(event.target.repetition.value),
    //         set_weight: this.state.set_weight,
    //         set_repetition: this.state.set_repetition,
    //     }
    //     console.log(updatedExerciseSet)
    //     console.log(this.props)
    //     this.props.handleClickUpdate(updatedExerciseSet)
        
    // }
    renderExerciseSetState() {
        const { edit, exerciseSet } = this.props
        const vol = (exerciseSet.set_weight * exerciseSet.set_repetition)

        if (edit) {
            return (
               <div className='SetItemUpdateForm'>
                <form
                className='SetItemUpdateForm'
                // onInput={event => this.handleFormChange(event)}
                >
                {/* <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div> */}
                <div className='weight'>
                    <label htmlFor='SetItem__weight'>
                        Weight
                    </label>
                    <input
                        required
                        name='set_weight'
                        id='SetItem__weight'
                        defaultValue={exerciseSet.set_weight}
                        onChange={this.onWeightInputChange.bind(this)}
                        >
                        
                    </input>
                    <span> lbs </span>
                </div>
                <div className='repetition'>
                    <label htmlFor='SetItem__repetition'>
                        Reps
                    </label>
                    <input
                        required
                        name='set_repetition'
                        type='integer'
                        id='SetItem__repetition'
                        defaultValue={exerciseSet.set_repetition}
                        onChange={this.onRepetitionInputChange.bind(this)}
                        >
                    </input>
                </div>
                <Button
                    className='test_button'
                    type='submit'
                    // onClick={this.handleClickUpdate}>
                    >
                    âœ” 
                </Button>
            </form>
            </div>
            )
        } else {
            return (
                <div className='ExerciseSet__item'>
                    <span>Set #</span>
                    <span>{exerciseSet.set_weight} lbs </span>
                    <span>{exerciseSet.set_repetition} reps </span>
                    <span>Vol: {vol} lbs</span>
                </div>
            )
        }
    }
    
    render() {

        return (
            <div className='ExerciseSet__item'>
                {this.renderExerciseSetState()}
            </div>
        )
    }
}