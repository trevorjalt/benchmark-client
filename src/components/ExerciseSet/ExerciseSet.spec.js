import React from 'react'
import { WorkoutProvider } from '../../contexts/WorkoutContext'
import ExerciseSet from './ExerciseSet'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`ExerciseSet Component`, () => {
    const props = {
        exerciseSet: {
            id: 1,
            set_weight: 100,
            set_repetition: 5,
            exercise_id: 1,
        }
    }
    it (`renders without crashing`, () => {
        const wrapper = 
            shallow (
                <WorkoutProvider>
                    <ExerciseSet {...props}/>
                </WorkoutProvider>
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})