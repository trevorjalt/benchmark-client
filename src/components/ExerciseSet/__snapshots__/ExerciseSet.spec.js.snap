// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseSet Component renders without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "exerciseList": Array [],
      "exerciseSetList": Array [],
      "isLoggedIn": null,
      "newWorkoutItem": Object {},
      "onDeleteExercise": [Function],
      "onDeleteExerciseSet": [Function],
      "onDeleteWorkout": [Function],
      "onUpdateExercise": [Function],
      "onUpdateExerciseSet": [Function],
      "setError": [Function],
      "setExerciseList": [Function],
      "setExerciseSetList": [Function],
      "setIsLoggedIn": [Function],
      "setWorkoutList": [Function],
      "touched": undefined,
      "workoutList": Array [],
    }
  }
>
  <ExerciseSet
    exerciseSet={
      Object {
        "exercise_id": 1,
        "id": 1,
        "set_repetition": 5,
        "set_weight": 100,
      }
    }
  />
</ContextProvider>
`;
